test/draw_polygon.test.js

```diff
@@ -0,0 +1,428 @@
+  t.deepEqual(context.ui.queueMapClasses.getCall(0).args, [{ mouse: 'add' }],
+    'ui.queueMapClasses received correct arguments');
+  t.equal(context.ui.setActiveButton.callCount, 1, 'ui.setActiveButton called');
+  t.deepEqual(context.ui.setActiveButton.getCall(0).args, ['polygon'],
+    'ui.setActiveButton received correct arguments');
+
+  t.equal(lifecycleContext.on.callCount, 4, 'this.on called');
+  t.ok(lifecycleContext.on.calledWith('mousemove', CommonSelectors.true));
+  t.ok(lifecycleContext.on.calledWith('click', CommonSelectors.true));
+  t.ok(lifecycleContext.on.calledWith('keyup', CommonSelectors.isEscapeKey));
+  t.ok(lifecycleContext.on.calledWith('keyup', CommonSelectors.isEnterKey));
+
+  setTimeout(() => {
+    t.equal(context.map.doubleClickZoom.disable.callCount, 1);
+    t.end();
+  }, 10);
+});
+
+test('draw_polygon stop with valid polygon', t => {
+  const context = createMockContext();
+  const mode = drawPolygonMode(context);
+
+  // Fake a valid polygon
+  context._test.polygon.isValid = () => true;
+
+  mode.stop.call();
+  t.equal(context.ui.setActiveButton.callCount, 1, 'ui.setActiveButton called');
+  t.deepEqual(context.ui.setActiveButton.getCall(0).args, [],
+    'ui.setActiveButton received correct arguments');
+  t.equal(context.store.delete.callCount, 0, 'store.delete not called');
+
+  t.end();
+});
+
+test('draw_polygon stop with invalid polygon', t => {
+  const context = createMockContext();
+  const mode = drawPolygonMode(context);
+
+  // Fake an invalid polygon
+  context._test.polygon.isValid = () => false;
+
+  mode.stop.call();
+  t.equal(context.ui.setActiveButton.callCount, 1, 'ui.setActiveButton called');
+  t.deepEqual(context.ui.setActiveButton.getCall(0).args, [],
+    'ui.setActiveButton received correct arguments');
+  t.equal(context.store.delete.callCount, 1, 'store.delete called');
+  t.deepEqual(context.store.delete.getCall(0).args, [
+    [context._test.polygon.id],
+    { silent: true }
+  ], 'store.delete received correct arguments');
+
+  setTimeout(() => {
+    t.equal(context.map.doubleClickZoom.enable.callCount, 1);
+    t.end();
+  }, 10);
+});
+
+test('draw_polygon render, active, with only two vertices', t => {
+  const context = createMockContext();
+  const mode = drawPolygonMode(context);
+
+  const memo = [];
+  const geojson = {
+    type: 'Feature',
+    properties: {
+      id: context._test.polygon.id
+    },
+    geometry: {
+      type: 'Polygon',
+      coordinates: [[[0, 0], [0, 10]]]
+    }
+  };
+  mode.render(geojson, x => memo.push(x));
+  t.equal(memo.length, 1);
+  t.deepEqual(memo[0], {
+    type: 'Feature',
+    properties: {
+      id: context._test.polygon.id,
+      active: 'true',
+      meta: 'feature'
+    },
+    geometry: {
+      type: 'LineString',
+      coordinates: [[0, 0], [0, 10]]
+    }
+  }, 'a line string is sent to the callback');
+  t.end();
+});
+
+test('draw_polygon render, inactive, with only two vertices', t => {
```

Comment:
This is inactive because there is no `id` not because the feature only has two vertices.